FreeRTOS_UDP_IP.c

```diff
@@ -319,19 +319,27 @@ BaseType_t xProcessReceivedUDPPacket( NetworkBufferDescriptor_t * pxNetworkBuffe
     {
         if( pxSocket != NULL )
         {
-            if( xCheckRequiresARPResolution( pxNetworkBuffer ) == pdTRUE )
+            if( *ipLOCAL_IP_ADDRESS_POINTER != 0U )
```

Comment:
I would like it more if we can move this check to `xCheckRequiresARPResolution` [here](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/ab0d4f84beee3acce2097020ef631169ac55f618/FreeRTOS_ARP.c#L404).

Because, the function `xCheckRequiresARPResolution` is supposed to return `pdFALSE`  if ARP resolution should not be done. And when it returns a `pdFALSE`, we'll just update the ARP cache entry's age and move on to proceed with processing of the packet.

What do you think @htibosch?

Reply:
During a DHCP negotiation, ARP does not play a role, and therefore I would not even call `xCheckRequiresARPResolution()`.
It doesn't really matter much, because `vARPRefreshCacheEntryAge()` would probably not find the IP-address to be refreshed.

Reply:
Very well. I am satisfied with that argument.

Thank you @htibosch. Would you mind making the change to unit-tests or shall I do that on your behalf?